internal/controller/pipelinerollout/pipelinerollout_controller.go

```diff
@@ -563,10 +563,16 @@ func (r *PipelineRolloutReconciler) processPipelineStatus(ctx context.Context, p
 
 	// Only fetch the latest pipeline object while deleting the pipeline object, i.e. when pipelineRollout.DeletionTimestamp.IsZero() is false
 	if existingPipelineDef == nil {
-		pipelineDef, err := r.makePromotedPipelineDefinition(ctx, pipelineRollout)
+		// determine name of the promoted Pipeline
+		pipelineName, err := progressive.GetChildName(ctx, pipelineRollout, r, common.LabelValueUpgradePromoted, r.client, true)
 		if err != nil {
-			return err
+			return fmt.Errorf("Unable to process pipeline status: err=%s", err)
 		}
+		pipelineDef := &unstructured.Unstructured{}
+		pipelineDef.SetGroupVersionKind(numaflowv1.PipelineGroupVersionKind)
+		pipelineDef.SetNamespace(pipelineRollout.Namespace)
+		pipelineDef.SetName(pipelineName)
+
```

Comment:
This is admittedly not related to the rest of the PR, but is another bug I found, and I wanted to be able to do comprehensive testing of everything together, which is why I kept it in one PR. The previous call to `makePromotedPipelineDefinition()` doesn't work in the case of the ISBServiceRollout having already been deleted. So, I didn't want to have to depend on that function as part of processing Pipeline status.